### PR TITLE
Add Fedora 37 image with gcc12

### PR DESCRIPTION
**Pull request by osteffenrh**
Add a Fedora 37 bases image that provides gcc 12, Python 3.11

# Description

Add a new Linux image, based on Fedora 37. It provides a gcc 12 toolchain.

Issue #59 

### Containers Affected

New Fedora 37 container.

### CODE REVIEW

**Comment by osteffenrh** [comment]
@vlsunil, you can try your PR with the image below until this PR is merged.

`docker pull ghcr.io/osteffenrh/edk2-containers/fedora-37-test:latest`

File: Fedora-37/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        gcc-${GCC_VERSION} \
+        gcc-aarch64-linux-gnu-${GCC_VERSION_CROSS} \
+        gcc-arm-linux-gnu-${GCC_VERSION_CROSS} \
+        gcc-riscv64-linux-gnu-${GCC_VERSION_CROSS} \
+        git \
+        libX11-devel \
+        libXext-devel \
+        libuuid-devel \
+        make \
+        nuget \
```
Comment (cfernald) [inline Fedora-37/Dockerfile:42]: I don't think it's a problem, but any reason 37 doesn't have PowerShell like 35? I believe it is used in some azure pipelines.

**Review by cfernald — APPROVED** [review]

**Comment by osteffenrh** [comment]
Thanks for the review.  Finally got around to look at this again.

Updated:

- Add powershell back in
- switch base image to Fedora 37 (not minimal!)
  The minimal image comes with microdnf, with is a pain to use (slow, no support for http URLs,...). The size difference does not matter (~50MB) compared to the final image size.

**Comment by osteffenrh** [comment]
Test run with the EDK2 CI with this image: https://github.com/tianocore/edk2/pull/4108

**Comment by osteffenrh** [comment]
CI with GCC12 works.

@cfernald, could you have a look again? Just making sure the latest edits are ok.

**Review by cfernald — APPROVED** [review]

**Pull request merged**